server/src/main/kotlin/com/kamelia/hedera/plugins/Authentication.kt

```diff
@@ -20,15 +20,11 @@ fun Application.configureAuthentication() {
     install(Authentication) {
         configureJWT(AuthJwt, Environment.secretAccess) { call, _ ->
             val token = call.getHeader(HttpHeaders.Authorization).replace("Bearer ", "")
-            val session = SessionManager.verify(token)
-            requireNotNull(session)
-            requireNotNull(session.user)
-            handleForceChangePassword(session.user, call)
-            UserPrincipal(session.sessionId, session.user, token)
+            val userState = SessionManager.verify(token) ?: throw ExpiredOrInvalidTokenException()
+            handleForceChangePassword(userState, call)
+            UserPrincipal(userState.uuid, userState, token)
         }
-        configureJWT(RefreshJwt, Environment.secretRefresh) { call, cred ->
-            val token = call.getHeader(HttpHeaders.Authorization).replace("Bearer ", "")
-            SessionManager.verifyRefresh(token)
+        configureJWT(RefreshJwt, Environment.secretRefresh) { _, cred ->
             JWTPrincipal(cred.payload)
```

Comment:
we actually need to verify the refresh token (that it's the current last one for the session)